kbase-extension/static/kbase/js/widgets/appWidgets2/validation.js

```diff
@@ -150,85 +150,123 @@ define([
             };
         }
 
-        function getObjectInfo(workspaceId, objectName, authToken, serviceUrl) {
+        /**
+         *
+         * @param {number} workspaceId
+         * @param {Array<string>} objectNames
+         * @param {string} authToken
+         * @param {string} serviceUrl
+         * @returns Promise resolving into an Array of strings, in the same order as the given objectNames
+         */
+        function getObjectTypes(workspaceId, objectNames, authToken, serviceUrl) {
             const workspace = new Workspace(serviceUrl, {
                 token: authToken,
             });
 
             return workspace
                 .get_object_info_new({
-                    objects: [{ wsid: workspaceId, name: objectName }],
+                    objects: objectNames.map((name) => ({ wsid: workspaceId, name })),
                     ignoreErrors: 1,
                 })
                 .then((data) => {
-                    if (data[0]) {
-                        return serviceUtils.objectInfoToObject(data[0]);
```

Comment:
This isn't really necessary, as only the object types are used downstream. It's probably not hurting much, but putting 1000s of objects through it might cause some delay, so I just whittled it down to some simply type formatting and returning an array of either type strings or `null`s.